packages/js/product-editor/src/components/schedule-publish-modal/schedule-publish-modal.tsx

```diff
@@ -67,7 +67,11 @@ export function SchedulePublishModal( {
 
 				<DateTimePicker
 					currentDate={ date }
-					onChange={ handleDateTimePickerChange }
+					onChange={ ( val ) => {
+						if ( val ) {
+							handleDateTimePickerChange( val );
+						}
+					} }
```

Comment:
```suggestion
					onChange={ ( val ) => {
						handleDateTimePickerChange( val ?? '' );
					} }
```
I didn't explicitly test this one, but I believe the if statement would cause the same issue as the password field.

Reply:
Having said that, maybe we should update the `handleDateTimePickerChange` function instead.

Reply:
Fixed with e1f833ffa51d6d0e7100adcce0347930fee5be5f